Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -1503,13 +1503,14 @@ NiftiImageIO::WriteImageInformation()
   //                   dim[2] is required for 2-D volumes,
   //                   dim[3] for 3-D volumes, etc.
   this->m_NiftiImage->nvox = 1;
-  // Spacial dims in ITK are given in mm.
+  // Spatial dims in ITK are given in mm.
   // If 4D assume 4thD is in SECONDS, for all of ITK.
   // NOTE: Due to an ambiguity in the nifti specification, some developers
   // external tools believe that the time units must be set, even if there
-  // is only one dataset.  Having the time specified for a purly spatial
+  // is only one dataset.  Having the time specified for a purely spatial
   // image has no consequence, so go ahead and set it to seconds.
-  this->m_NiftiImage->xyz_units = static_cast<int>(NIFTI_UNITS_MM | NIFTI_UNITS_SEC);
+  this->m_NiftiImage->xyz_units = static_cast<int>(NIFTI_UNITS_MM);
+  this->m_NiftiImage->time_units = static_cast<int>(NIFTI_UNITS_SEC);
```

Comment:
Just for my understanding, your pull request should ensure that `NiftiImageIO::SetImageIOMetadataFromNIfTI()` will store "10" in its dictionary, after it does `xyzt_units << SPACE_TIME_TO_XYZT(nim->xyz_units, nim->time_units)`, right? At 
https://github.com/InsightSoftwareConsortium/ITK/blob/7f1c5846087f463ecb7875fb36053096911df92e/Modules/IO/NIFTI/src/itkNiftiImageIO.cxx#L948-L949

Reply:
I believe it should store whatever the metadata was in the original NIFTI file. But when the image is read, any the spacing and time should be adjusted to MM and SEC in the ITK image. Then when we write, the image should have units MM and SEC (xyzt_units=10) regardless of what the input was.

I think the metadata is being written correctly now because when a file is written, it has xyzt_units=10 and not 2. But I just noticed the time scaling may not be happening correctly:

https://github.com/InsightSoftwareConsortium/ITK/blob/7f1c5846087f463ecb7875fb36053096911df92e/Modules/IO/NIFTI/src/itkNiftiImageIO.cxx#L1325-L1336

I'm not sure why it's even calling the macros here, `nim->xyz_units` and `nim->time_units` are already masked versions of the header's `xyzt_units`.

I'll extend the tests to catch this. I will need to make a new test image with ms time units.

Reply:
@cookpa Cool! By the way, if you would still like to extend the tests as part of this pull request, you might want to click "Convert to draft", to indicate that it is still in progress. Otherwise, your PR might accidentally get merged before you are finished!

Reply:
Now updated to use this->m_NiftiImage->xyz_units for space and (this->m_NiftiImage->time_units for time